### PR TITLE
VIH-7389 participant stuck in waiting room event hub disconnect

### PR DESCRIPTION
**Pull request by FarronF**
### JIRA link (if applicable) ###
VIH-7389

### Change description ###
When SignalR connection is interrupted disconnect from pexip to prevent users appearing in hearings without knowing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/services/events-hub.service.ts [hunk 1]
```diff
@@ -112,6 +111,7 @@ export class EventsHubService {
                 .then(() => {
                     this.logger.info('[EventsService] - Successfully connected to EventHub');
                     this._reconnectionAttempt = 0;
+                    this.onEventHubReconnected();
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/events-hub.service.ts:114]: When reconnecting this event was not being emitted previously.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.eventhub-events.spec.ts [hunk 2]
```diff
@@ -94,25 +111,38 @@ describe('WaitingRoomComponent EventHub Call', () => {
...
+
+        TestBed.configureTestingModule({
+            declarations: [WRTestComponent],
+            // declarations: [ErrorComponent, ContactUsFoldingComponent, Mock1Component, Mock2Component, TranslatePipeMock],
+            // imports: [
+            //     RouterTestingModule.withRoutes([
+            //         { path: 'testUrl-test-error1', component: Mock1Component },
+            //         { path: 'testUrl-test-error2', component: Mock2Component }
+            //     ])
+            // ],
```
Comment (mulvaneyjHMCTS) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.eventhub-events.spec.ts:123]: Remove commented code

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.video-call-events.spec.ts [hunk 3]
```diff
@@ -120,6 +120,8 @@ describe('WaitingRoomComponent Video Call', () => {
     it('should define outgoing stream when video call has been setup', () => {
         const outgoingStream = <any>{};
         const payload = new CallSetup(outgoingStream);
+        // component.eventServiceConnected = true;
+
```
Comment (mulvaneyjHMCTS) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.video-call-events.spec.ts:124]: Remove

**Review by mulvaneyjHMCTS — COMMENTED** [review]
Looks LGMT, two small bits

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 4]
```diff
@@ -1190,4 +1206,9 @@ export abstract class WaitingRoomBaseDirective {
     switchStreamWindows(): void {
         this.streamInMain = !this.streamInMain;
     }
+
+    async callAndUpdateView(): Promise<void> {
```
Comment (mulvaneyjHMCTS) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:1210]: Maybe change to callAndUpdateVideo?

File: VideoWeb/VideoWeb/ClientApp/src/app/services/events.service.ts [hunk 5]
```diff
@@ -238,6 +238,14 @@ export class EventsService {
         this._handlersRegistered = false;
     }
 
+    get eventHubIsConnected(): boolean {
+        return this.eventsHubService.isConnectedToHub;
+    }
+
+    onEventsHubReady(): Observable<any> {
+        return this.eventsHubService.onEventsHubReady;
+    }
```
Comment (mulvaneyjHMCTS) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/events.service.ts:247]: Is it necessary to have both?

Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/events.service.ts:247]: Yes, I checked again and I think having both is needed.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.eventhub-events.spec.ts [hunk 6]
```diff
@@ -59,14 +60,30 @@ import { Room } from '../../../shared/models/room';
 import { RoomTransfer } from '../../../shared/models/room-transfer';
 import { ElementRef } from '@angular/core';
 import { VhToastComponent } from 'src/app/shared/toast/vh-toast.component';
-import { ConsultationInvitation } from '../../services/consultation-invitation.service';
+import { ConsultationInvitation, ConsultationInvitationService } from '../../services/consultation-invitation.service';
 import { Participant } from 'src/app/shared/models/participant';
+import { CallSetup } from '../../models/video-call-models';
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.eventhub-events.spec.ts:65]: looks like call setup is actually not used in this test file. There are quite a lot of unused imports/variables in this file it seems, might be worth removing them?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts [hunk 7]
```diff
@@ -427,72 +429,110 @@ describe('WaitingRoomComponent message and clock', () => {
...
+            });
 
-        expect(videoCallService.makeCall).toHaveBeenCalledWith(
-            room.pexip_node,
-            room.participant_join_uri,
-            room.tile_display_name,
-            component.maxBandwidth
-        );
-    });
+            it('should not call when event hub is not connected and token is not set', () => {});
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts:447]: has this been left by mistake? Looks to be an empty test block

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts [hunk 8]
```diff
@@ -427,72 +429,110 @@ describe('WaitingRoomComponent message and clock', () => {
...
-            participant_join_uri: 'patjoinuri',
-            pexip_node: 'sip.test.node',
-            display_name: 'foo',
-            tile_display_name: `I1;Interpreter1;${component.participant.id}`
-        });
-        videoCallService.retrieveWitnessInterpreterRoom.and.resolveTo(room);
-
-        await component.call();
+            it('should not call when event hub is not connected and token is set', () => {
+                component.token = new TokenResponse();
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts:450]: This test doesn't have any assertions, is that intentional?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts (hunk 7 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts:447]: The setup for this are handled by the beforeEach of the describe. I'll at least add a comment but I could see an argument that setup would be clearer if it was handled individually.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts (hunk 8 above)
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/tests/waiting-room-base.component.non-event.spec.ts:450]: I expect all these to fail the same way so the assertion is in the afterEach()

**Review by kevin-hmcts — APPROVED** [review]
LGTM

**Pull request merged**